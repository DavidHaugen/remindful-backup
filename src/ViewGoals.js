import React, { Component } from 'react';
import {Link} from 'react-router-dom'
import './styles/ViewGoals.css'
import Goal from './Goal'

class ViewGoals extends Component {
  createList(goals){
    return goals.map((goal, i) => <Goal key={i} goal={goal} />)
  };

  render(){
    if(this.props.goals.length){

      return(
        <div>
          <h1>Here are your goals</h1>
          <ul>
            {this.createList(this.props.goals)}
          </ul>
          <Link to='/add-goal' >
            <button>Add goal</button>
          </Link>
        </div>
      )

      }else return(
        <div>
          <h1>It looks like you haven't added any goals yet. Time to get started!</h1>
          <Link to='/add-goal' >
            <button>Add goal</button>
          </Link>
        </div>
      )
  }
}

export default ViewGoals