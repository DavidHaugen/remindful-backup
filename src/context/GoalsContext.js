import React, { Component } from 'react'


const GoalsContext = React.createContext({
  goals: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setGoals: () => {}
})

export default GoalsContext

export class GoalsProvider extends Component {
  state = {
    goals: [],
    error: null,
  };

  setGoals = goals => {
    this.setState({ goals })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      goals: this.state.goals,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGoals: this.setGoals
    }

    return (
      <GoalsContext.Provider value={value}>
        {this.props.children}
      </GoalsContext.Provider>
    )
  }
}