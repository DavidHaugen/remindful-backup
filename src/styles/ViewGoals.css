.listContainer{
  list-style: none;
width: 90%;
  min-height: 40vh;
  border: 2px solid #0A0908;
  border-radius: 5px;
  padding: 0 10px 40px 10px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 25px;
  background-color: #FAFAFA;
  font-family: 'Rubik', sans-serif;
}

.goal{
  text-decoration: none;
  color: #0A0908;
  text-align: left;
  padding: 5px 20px;
  border-radius: 5px;
  cursor: pointer;
  height: 60px;
  display: flex;
  align-items: center;
  font-size: 20px;
  margin-top: 10px;
}

.goal:nth-child(even){
  background-color: #dce3e9;
}

.goal:nth-child(odd){
  background-color: rgb(199, 218, 198)
}

.button{
  color: #93394B;
  background-color: #FAFAFA;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  font-size: 35px;
  text-align: center;
  height:40px;
  padding: 0
}

.bigButton{
  font-size: 60px;
}

.addLink{
  height:35px;
  width:35px;
  font-size: 35px;
  border-radius: 50%;
  margin-bottom:4px;
}

.listTitle{
  display: inline-block;
}

.listHeader{
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 20px 0 20px 0
}

.listHeader h1{
  margin: 0;
  font-family: 'Karla', sans-serif;
  font-size: 36px;
  color: #0A0908;
  margin-right: 30px;
}