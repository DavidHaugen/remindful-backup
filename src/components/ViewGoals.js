import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import '../styles/ViewGoals.css'
import Goal from './Goal'
import GoalsContext from '../context/GoalsContext';
import '../styles/main.css'


class ViewGoals extends Component {

  static contextType = GoalsContext

  createList(goals){
    return goals.map((goal, i) => <Goal key={i} goal={goal} history={this.props.history}/>)
  };

  render(){
    if(this.context.loading){
      return(
        <div className="main" role="main">
          <div className="lds-grid"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
        </div>
      )
    } else{
    
    if(this.context.goals.length > 0){

      return(
        <div className="main" role="main">
          <div className="wrapper">
            <div role='alert'>
              {this.context.error && <p className='red'>{this.context.error}</p>}
            </div>
            <ul className="listContainer">
              <div className="listHeader">
                <h1 className="listTitle">Your goals</h1>
                <Link to='/add-goal' className="addLink">
                  <button className="button"><i className="fas fa-plus-circle"></i></button>
                </Link>
              </div>
              {this.createList(this.context.goals)}
            </ul>
          </div>
        </div>
      )

      }else {return(
        <div className="main" role="main">
          <div className="wrapper">
            <h1>It looks like you haven't added any goals yet. Click the button below to add your first goal.</h1>
            <Link to='/add-goal'>
              <button className="button"><i className="fas fa-plus-circle bigButton"></i></button>
            </Link>
          </div>
        </div>
      )
      }
    }
  }
}

export default ViewGoals