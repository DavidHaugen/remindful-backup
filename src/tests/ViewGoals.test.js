import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import ViewGoals from '../components/ViewGoals'
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import GoalsContext from '../context/GoalsContext';

describe('<ViewGoals />', () => {
  it('Renders without crashing', () => {
    mount(<MemoryRouter><ViewGoals/></MemoryRouter>);
  });
  it('renders the UI as expected when no goals are present', () => {
    const tree = renderer
      .create(<MemoryRouter><ViewGoals/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  it('renders the UI as expected when no goals are present', () => {
    const tree = renderer
      .create(<MemoryRouter><ViewGoals/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  it('renders the UI as expected when there are goals present', () => {
    const state ={ goals: [{complete: false, id: 1, name: 'test'}, {complete: false, id: 2, name: 'test2'}]}
    const tree = renderer
      .create(<MemoryRouter><GoalsContext.Provider value={state}
        ><ViewGoals/></GoalsContext.Provider></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  it('Renders a list of goals with the correct number of goals', () => {
    const state ={ goals: [{complete: false, id: 1, name: 'test'}, {complete: false, id: 2, name: 'test2'}]}
    const wrapper = mount(<MemoryRouter><GoalsContext.Provider value={state}
    ><ViewGoals/></GoalsContext.Provider></MemoryRouter>);
    const items = wrapper.find('.goal');
    expect(items.length).toEqual(2);
    });

});