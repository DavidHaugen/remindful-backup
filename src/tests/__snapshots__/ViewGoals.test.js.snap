// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewGoals /> renders the UI as expected when no goals are present 1`] = `
<div
  className="main"
  role="main"
>
  <div
    className="wrapper"
  >
    <h1>
      It looks like you haven't added any goals yet. Click the button below to add your first goal.
    </h1>
    <a
      href="/add-goal"
      onClick={[Function]}
    >
      <button
        className="button"
      >
        <i
          className="fas fa-plus-circle bigButton"
        />
      </button>
    </a>
  </div>
</div>
`;

exports[`<ViewGoals /> renders the UI as expected when no goals are present 2`] = `
<div
  className="main"
  role="main"
>
  <div
    className="wrapper"
  >
    <h1>
      It looks like you haven't added any goals yet. Click the button below to add your first goal.
    </h1>
    <a
      href="/add-goal"
      onClick={[Function]}
    >
      <button
        className="button"
      >
        <i
          className="fas fa-plus-circle bigButton"
        />
      </button>
    </a>
  </div>
</div>
`;

exports[`<ViewGoals /> renders the UI as expected when there are goals present 1`] = `
<div
  className="main"
  role="main"
>
  <div
    className="wrapper"
  >
    <div
      role="alert"
    />
    <ul
      className="listContainer"
    >
      <div
        className="listHeader"
      >
        <h1
          className="listTitle"
        >
          Your goals
        </h1>
        <a
          className="addLink"
          href="/add-goal"
          onClick={[Function]}
        >
          <button
            className="button"
          >
            <i
              className="fas fa-plus-circle"
            />
          </button>
        </a>
      </div>
      <div
        className="goal"
        onClick={[Function]}
      >
        <li
          style={
            Object {
              "textDecoration": "none currentcolor solid",
            }
          }
        >
          test
        </li>
      </div>
      <div
        className="goal"
        onClick={[Function]}
      >
        <li
          style={
            Object {
              "textDecoration": "none currentcolor solid",
            }
          }
        >
          test2
        </li>
      </div>
    </ul>
  </div>
</div>
`;
